src/Protocol.java

```diff
@@ -0,0 +1,11 @@
+import java.net.InetAddress;
+
+public interface Protocol {
+    public void pack(String msg, String hostname, int port);
```

Comment:
let's add javadoc to the interfaces

Reply:
I don't get what are the differences between these 3 pack methods. Maybe use more detailed method name?